tracer/src/Datadog.Trace/AppSec/Waf/Initialization/WafConfigurator.cs

```diff
@@ -23,10 +24,17 @@ internal static class WafConfigurator
 
         internal static InitializationResult Configure(string rulesFile, WafNative wafNative, Encoder encoder, string obfuscationParameterKeyRegex, string obfuscationParameterValueRegex)
         {
+            if (wafNative.ExportErrorHappened)
+            {
+                Log.Error("Waf couldn't initialize properly because of missing methods in native library, please check the version of libddwaf");
+                return InitializationResult.FromWrongWafVersion();
+            }
+
             var argCache = new List<Obj>();
             var configObj = GetConfigObj(rulesFile, argCache, encoder);
             if (configObj == null)
             {
+                Log.Error("Waf couldn't initialize properly because of an unusable rule file, please check the submitted rulefile");
```

Comment:
Same comment here, is there any way to have something actionable for the user in the logs? e.g. check for multiple tracer versions etc etc "if you manually provided a rule file, make sure it's valid for the latest version of the Waf" etc

Reply:
fixed in https://github.com/DataDog/dd-trace-dotnet/pull/3251/commits/9fe3b5df8b64a610153f2a3ef1677b24a138514d